src/main/java/com/minecolonies/coremod/entity/pathfinding/PathJobCanSee.java

```diff
@@ -0,0 +1,87 @@
+package com.minecolonies.coremod.entity.pathfinding;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RayTraceContext;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Pathing job for moving into vision of the given entity
+ */
+public class PathJobCanSee extends AbstractPathJob
+{
+    /**
+     * The entity to see
+     */
+    private final LivingEntity lookTarget;
+
+    /**
+     * The entity to move
+     */
+    private final LivingEntity searchingEntity;
+
+    public PathJobCanSee(
+      final LivingEntity searchingEntity,
+      final LivingEntity lookTarget,
+      final World world,
+      @NotNull final BlockPos start, final int range)
+    {
+        super(world, start, start, range, searchingEntity);
+
+        this.searchingEntity = searchingEntity;
+        this.lookTarget = lookTarget;
+    }
+
+    @Override
+    protected double computeHeuristic(final BlockPos pos)
+    {
+        return searchingEntity.getPosition().manhattanDistance(pos);
+    }
+
+    @Override
+    protected boolean isAtDestination(final Node n)
+    {
+        if (Math.abs(start.getY() - n.pos.getY()) > 6)
```

Comment:
I don't see a problem going up, I think the problem is only going down imo

Reply:
but some might want to station them not on the highest but inbetween somewhere

Reply:
Let's say I did that. I'd be fine for them to go a bit up to hit the target, but not to go much further down (compared to the guarding position)